File: kafka-helper-domain/src/main/java/com/hg/kafka/helper/domain/entity/MessageLog.java
package com.hg.kafka.helper.domain.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.Date;

@Document("message_log")
@Data
public class MessageLog {
    @Id
    public String id;
    @Field("msg_id")
    private String messageId;
    @Field("msg")
    private String messageJson;
    @Field("topic")
    private String topic;
    @Field("env")
    private String environment;
    @Field("created_at")
    private Date createdAt;
}

File: kafka-helper-domain/src/main/kotlin/com/hg/kafka/helper/domain/entity/MessageLog.kt
package com.hg.kafka.helper.domain.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field
import java.util.*

@Document("message_log")
data class MessageLog(
        @Id var id: String? = null,
        @Field("msg_id") var messageId: String,
        @Field("msg") var messageJson: String,
        @Field("topic") var topic: String,
        @Field("env") var environment: String,
        @Field("created_at") var createdAt: Date = Date()
)